tests/plugins/webp-uploads/load-tests.php

```diff
@@ -632,43 +665,86 @@ public function test_it_should_preserve_jpeg_subsizes_using_transform_filter():
+				// therefore the one that does support modern images needs to be first in this list.
+				array_unshift( $editors, 'WP_Image_Doesnt_Support_Modern_Images' );
+				return $editors;
 			}
 		);
 
-		$this->assertTrue( wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) );
+		$this->assertTrue( wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) );
 
+		// Ensure the output type is WebP for this test.
+		$this->set_image_output_type( $image_type );
 		$attachment_id = self::factory()->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/data/images/leaves.jpg' );
 		$metadata      = wp_get_attachment_metadata( $attachment_id );
 
 		$this->assertArrayHasKey( 'sources', $metadata );
 		$this->assertIsArray( $metadata['sources'] );
 
 		$this->assertImageNotHasSource( $attachment_id, 'image/jpeg' );
-		$this->assertImageHasSource( $attachment_id, 'image/webp' );
+		$this->assertImageHasSource( $attachment_id, 'image/' . $image_type );
 
 		$this->assertImageNotHasSizeSource( $attachment_id, 'thumbnail', 'image/jpeg' );
-		$this->assertImageHasSizeSource( $attachment_id, 'thumbnail', 'image/webp' );
+		$this->assertImageHasSizeSource( $attachment_id, 'thumbnail', 'image/' . $image_type );
 	}
 
 	/**
-	 * Replace the featured image to WebP when requesting the featured image
+	 * Replace the featured image to the proper type when requesting the featured image.
+	 *
+	 * @dataProvider data_provider_supported_image_types
+	 *
+	 * @test
+	 * @param string $image_type
 	 */
-	public function test_it_should_replace_the_featured_image_to_webp_when_requesting_the_featured_image(): void {
+	public function test_it_should_replace_the_featured_image_to_webp_when_requesting_the_featured_image( string $image_type ): void {
+		$mime_type = 'image/' . $image_type;
+
+		// Skip this test if the image editor doesn't support the image type.
+		if ( ! webp_uploads_mime_type_supported( $mime_type ) ) {
+				$this->markTestSkipped( 'The image editor does not support the image type: ' . $image_type );
+		}
+
+		// Set the image output format.
+		$this->set_image_output_type( $image_type );
+
 		$attachment_id = self::factory()->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/data/images/paint.jpeg' );
 		$post_id       = self::factory()->post->create();
 		set_post_thumbnail( $post_id, $attachment_id );
 
 		$featured_image = get_the_post_thumbnail( $post_id );
-		$this->assertMatchesRegularExpression( '/<img .*?src=".*?\.webp".*>/', $featured_image );
+		$this->assertMatchesRegularExpression( '/<img .*?src=".*?\.' . $image_type . '".*>/', $featured_image );
+	}
+
+	/**
+	 * Data provider for tests returns the supported image types to run the tests against.
+	 *
+	 * @return array<array<string>> An array of valid image types.
+	 */
+	public function data_provider_supported_image_types(): array {
+		return array(
+			array( 'webp' ),
+			array( 'avif' ),
```

Comment:
Can help when wanting to run some specific test data with `--filter`:


```suggestion
	 * @return array<string, array<string>> An array of valid image types.
	 */
	public function data_provider_supported_image_types(): array {
		return array(
			'webp' => array( 'webp' ),
			'avif' => array( 'avif' ),
```